boilerplate/_lib/ensure.sh

```diff
@@ -1,11 +1,14 @@
 #!/bin/bash
```

Comment:
IMO, this entire `ensure.sh` should really just go away. This is really what Makefile dependency chains are for. Why can't we run these commands under Makefile targets and have a proper chain?

For example:
```
operator-sdk-bin: /path/to/operator-sdk
    #install operator-sdk commands

opgenerate: operator-sdk-bin
    $(OPERATOR_SDK) generate $(OPERATOR_SDK_TARGETS)
```

Reply:
I did play with something [close to that](https://github.com/openshift/boilerplate/commit/689cbda3fc7649ea06e77d7ef343061de2bdabad#diff-6efb8dd653c9062efcfb26adab7756fc) at one point in this PR, and I'm okay going back in that direction if folks feel strongly about it. But it gets nowhere near eliminating `ensure.sh`, and IMO it shouldn't.

`make` simply isn't designed for complex branching/looping. Yes, it has conditionals, but they're awkward to use and read, especially when they need to be nested *. And while `ensure.sh` isn't super complicated, each of its cases has enough logic to be awkward if ported to native `make`.

\* Using bash conditionals in the target doesn't count. I don't object to a target that runs multiple commands, but I'm a hard no on "one-line shell scripts" embedded in `make`. For example, [this](https://github.com/openshift/must-gather-operator/blob/49dd9f1fa3321b9506478fcd56970635ecb871cb/functions.mk) has got to go.